src/DeploymentProvider/DeploymentHelper.ts

```diff
@@ -0,0 +1,82 @@
+        deployments.forEach(deployment => {
+            console.log("deployment:" + deployment);
+            console.log('deployment str: ' + JSON.stringify(deployment));
+            names.push(deployment.name);
+        });
+        return names;
+    }
+
+    public static async setActiveDeployment(client: AppPlatformManagementClient, params: ActionParameters) {
+        let appResource: Models.AppResource = {
+            properties: {
+                activeDeploymentName: params.DeploymentName
+            }
+        };
+        await client.apps.update(params.ResourceGroupName, params.ServiceName, params.AppName, appResource);
+        return;
+    }
+
+    public static async deploy(client: AppPlatformManagementClient, params: ActionParameters, sourceType: string, fileToUpload: string) {
+        let uploadResponse: Models.AppsGetResourceUploadUrlResponse = await client.apps.getResourceUploadUrl(params.ResourceGroupName, params.ServiceName, params.AppName);
+        await uploadFileToSasUrl(uploadResponse.uploadUrl, fileToUpload);
+        let transformedEnvironmentVariables = {};
+        if (params.EnvironmentVariables) {
+            core.debug("Environment variables modified.");
+            const parsedEnvVariables = parse(params.EnvironmentVariables);
+            //Parsed pairs come back as  {"key1":{"value":"val1"},"key2":{"value":"val2"}}
+            Object.keys(parsedEnvVariables).forEach(key => {
+                transformedEnvironmentVariables[key] = parsedEnvVariables[key]['value'];
+            });
+            core.debug('Environment Variables: ' + JSON.stringify(transformedEnvironmentVariables));
+        }
+        let deploymentResource: Models.DeploymentResource = {
+            properties: {
+                source: {
+                    relativePath: uploadResponse.relativePath,
+                    type: sourceType as Models.UserSourceType,
+                    version: params.Version
+                },
+                deploymentSettings: {
+                    jvmOptions: params.JvmOptions,
+                    netCoreMainEntryPath: params.DotNetCoreMainEntryPath,
+                    runtimeVersion: params.RuntimeVersion as Models.RuntimeVersion,
+                    environmentVariables: transformedEnvironmentVariables
+                }
+            }
+        }
+        core.debug("deploymentResource:" + JSON.stringify(deploymentResource));
+        const response = await client.deployments.createOrUpdate(params.ResourceGroupName, params.ServiceName, params.AppName, params.DeploymentName, deploymentResource);
+        core.debug("deployment response:\n" + response._response.bodyAsText);
+        return;
+    }
+
+    public static async deleteDeployment(client: AppPlatformManagementClient, params: ActionParameters) {
+        await client.deployments.deleteMethod(params.ResourceGroupName, params.ServiceName, params.AppName, params.DeploymentName);
```

Comment:
We need to check for the return values. 

It seems we ignores all the SDK api call return values. We need to check how we can propagate the error response from our RP to the GitHub Actions error output.